patches/server/0962-Log-warning-instead-of-exception-if-plugin-bootstrap.patch

```diff
@@ -0,0 +1,22 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: viktorhrekh <viktorhrekh@viesoft.dev>
+Date: Fri, 24 Feb 2023 19:51:13 +0200
+Subject: [PATCH] Log warning instead of exception if plugin bootstrap loads
+ another plugin
+
+
+diff --git a/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginParent.java b/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginParent.java
+index a0773c6d24de1f2ce1f0d949cba26b9997b696e6..95f9f31bc767478da3d762fdd4120cbeea2deb49 100644
+--- a/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginParent.java
++++ b/src/main/java/io/papermc/paper/plugin/provider/type/paper/PaperPluginParent.java
+@@ -172,9 +172,8 @@ public class PaperPluginParent {
+                     plugin = bootstrap.createPlugin(PaperPluginParent.this.context);
+                 }
+ 
+-                // Don't allow plugins to load plugins other than the one defined in main. This restriction might not be necessary.
+                 if (!plugin.getClass().isAssignableFrom(Class.forName(PaperPluginParent.this.description.getMainClass(), true, plugin.getClass().getClassLoader()))) {
+-                    throw new IllegalArgumentException("Plugin provided must be the same type as main defined in plugin configuration!");
++                    logger.warning("Plugin " + getMeta().getName() + " uses bootstrap that is returning another plugin.");
```

Comment:
Does this even have to be a warning? I would just log an info that contains the type of the provided plugin and something about it not matching the configured main class.